Schools mode share

mem48 opened the pull request:
Estimate of baseline cycling added.

Trying a test build

```
> targets::tar_make()
Skipping install of 'cyclestreets' from a github remote, the SHA1 (382d4b6d) has not changed since last install.
  Use `force = TRUE` to force installation
── Attaching packages ─────────────────────────────────────── tidyverse 1.3.2 ──
✔ ggplot2 3.4.2     ✔ purrr   1.0.1
✔ tibble  3.2.1     ✔ dplyr   1.1.2
✔ tidyr   1.3.0     ✔ stringr 1.5.0
✔ readr   2.1.4     ✔ forcats 1.0.0
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
Linking to GEOS 3.9.3, GDAL 3.5.2, PROJ 8.2.1; sf_use_s2() is TRUE
tmap mode set to interactive viewing
`summarise()` has grouped output by 'Local Authority'. You can override using
the `.groups` argument.
✔ skip target subpoints_origins
• start target param_file
• built target param_file [0 seconds]
✔ skip target subpoints_destinations
• start target parameters
• built target parameters [0 seconds]
• start target study_area
• built target study_area [0.88 seconds]
• start target zones
• built target zones [0.26 seconds]
• start target r_school
Getting the fastest routes for school journeys
2023-07-06 17:05:17 doing batch 1 of 1
Number of rows in batch: 1160
Looking in the file: tmp/batch_fastest_school_1_with_1160_routes_of_1160_rows.Rds
File does not exist
Starting routing
POSTing the request to create and start the job
Job id: 4815
No data returned yet. Trying again id 4815 every 10 seconds
Congrats, you data is ready
Reading in the following file:
C:\Users\earmmor\AppData\Local\Temp\RtmpMt8SQo/test.csv.gz
Rows: 1160 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): start_id, end_id, strategy, json
dbl (3): distance, time_seconds, calories
|
ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
0 routes removed
Error in sf_column %in% names(g) : 
  Join columns in `x` must be present in the data.
✖ Problem with `id`.
In addition: There were 13 warnings (use warnings() to see them)
Starting routing
POSTing the request to create and start the job
Job id: 4816
No data returned yet. Trying again id 4816 every 10 seconds
```

mem48 commented:
Seems to be stuck in an infinite loop

```
Rows: 1160 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): start_id, end_id, strategy, json
dbl (3): distance, time_seconds, calories
|
ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
0 routes removed
Error in sf_column %in% names(g) : 
  Join columns in `x` must be present in the data.
✖ Problem with `id`.
Starting routing
POSTing the request to create and start the job
Job id: 4918
No data returned yet. Trying again id 4918 every 10 seconds
Congrats, you data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\earmmor\AppData\Local\Temp\RtmpMt8SQo/test.csv.gz
Rows: 1160 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): start_id, end_id, strategy, json
dbl (3): distance, time_seconds, calories
\
ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
0 routes removed
Error in sf_column %in% names(g) : 
  Join columns in `x` must be present in the data.
✖ Problem with `id`.
Starting routing
POSTing the request to create and start the job
Job id: 4919
No data returned yet. Trying again id 4919 every 10 seconds
Congrats, you data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\earmmor\AppData\Local\Temp\RtmpMt8SQo/test.csv.gz
Rows: 1160 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): start_id, end_id, strategy, json
dbl (3): distance, time_seconds, calories
\
ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
0 routes removed
Error in sf_column %in% names(g) : 
  Join columns in `x` must be present in the data.
✖ Problem with `id`.
Starting routing
POSTing the request to create and start the job
Job id: 4920
No data returned yet. Trying again id 4920 every 10 seconds
Congrats, you data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\earmmor\AppData\Local\Temp\RtmpMt8SQo/test.csv.gz
Rows: 1160 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): start_id, end_id, strategy, json
dbl (3): distance, time_seconds, calories
/
ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
0 routes removed
Error in sf_column %in% names(g) : 
  Join columns in `x` must be present in the data.
✖ Problem with `id`.
Starting routing
POSTing the request to create and start the job
Job id: 4921
No data returned yet. Trying again id 4921 every 10 seconds
Congrats, you data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\earmmor\AppData\Local\Temp\RtmpMt8SQo/test.csv.gz
Rows: 1160 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): start_id, end_id, strategy, json
dbl (3): distance, time_seconds, calories
/
ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
0 routes removed
Error in sf_column %in% names(g) : 
  Join columns in `x` must be present in the data.
✖ Problem with `id`.
Starting routing
POSTing the request to create and start the job
Job id: 4922
No data returned yet. Trying again id 4922 every 10 seconds

```

Robinlovelace commented:
Yes that seems to be the issue... Will diagnose.

Robinlovelace commented:
Does indeed seem you are in a loop @mem48 ...

Robinlovelace commented on code/prep_school_data.R:
```diff
@@ -104,6 +103,126 @@ flow_jitter <- flow_jitter[,c("DataZone","SeedCode","count")]
 
-saveRDS(flow_jitter, "D:/University of Leeds/TEAM - Network Planning Tool - General/secure_data/schools/school_dl_sub30km.Rds")
+# Add on mode shares
+flow_jitter$lenght_km_str <- as.numeric(st_length(flow_jitter)) / 1000
```
Nitpick but please use consistent project style: `=`

Robinlovelace reviewed (changes requested):
I get this:

```
Error:
! Error running targets::tar_visnetwork()
  Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
  Debugging guide: https://books.ropensci.org/targets/debugging.html
  How to ask for help: https://books.ropensci.org/targets/help.html
  Last error: `path` does not exist: ‘NA’
```

Robinlovelace commented:
> Error in sf_column %in% names(g) : 
>   Join columns in `x` must be present in the data.
> ✖ Problem with `id`.

Fixed upstream: https://github.com/cyclestreets/cyclestreets-r/issues/64

Robinlovelace commented:
@mem48 you broke the build by putting scripts in the R folder.

Robinlovelace commented:
Progress here, the routing completed. Now hitting the following error in the `uptake_list_school` target:

```
 Last error: ℹ In argument: `pcycle_go_dutch = pcycle_go_dutch + (bicycle/all)`.
ℹ In group 1: `route_id = 19132`.
Caused by error:
! object 'bicycle' not found
```

![image](https://github.com/nptscot/npt/assets/1825120/01231032-34ff-4ff2-8a60-94a2319395f5)

mem48 commented:
New fail 

```
> targets::tar_make()
Skipping install of 'cyclestreets' from a github remote, the SHA1 (86901c4e) has not changed since last install.
  Use `force = TRUE` to force installation
── Attaching packages ─────────────────────────────────────── tidyverse 1.3.2 ──
✔ ggplot2 3.4.2     ✔ purrr   1.0.1
✔ tibble  3.2.1     ✔ dplyr   1.1.2
✔ tidyr   1.3.0     ✔ stringr 1.5.0
✔ readr   2.1.4     ✔ forcats 1.0.0
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
Linking to GEOS 3.9.3, GDAL 3.5.2, PROJ 8.2.1; sf_use_s2() is TRUE
✔ skip target subpoints_origins
✔ skip target param_file
✔ skip target subpoints_destinations
✔ skip target parameters
✔ skip target study_area
✔ skip target zones
• start target r_school
Getting the fastest routes for school journeys
2023-07-07 12:38:53 doing batch 1 of 1
Number of rows in batch: 1160
Looking in the file: tmp/batch_fastest_school_1_with_1160_routes_of_1160_rows.Rds
File does not exist
Starting routing
POSTing the request to create and start the job
Job id: 5003
Congrats, your data is ready
Reading in the following file:
C:\Users\earmmor\AppData\Local\Temp\Rtmp6b251j/test.csv.gz
Rows: 1160 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): start_id, end_id, strategy, json
dbl (3): distance, time_seconds, calories
|
ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
0 routes removed
5003: deleted
Saving tmp/batch_fastest_school_1_with_1160_routes_of_1160_rows.Rds to tmp
Saving tmp/od_school_1_with_1160_of_1160_rows.Rds to tmp
Combining results
Saving outputdata/2023-07-04/routes_max_dist_school_fastest.Rds
Getting the balanced routes for school journeys
2023-07-07 12:40:41 doing batch 1 of 1
Number of rows in batch: 1160
Looking in the file: tmp/batch_balanced_school_1_with_1160_routes_of_1160_rows.Rds
File does not exist
Starting routing
POSTing the request to create and start the job
Job id: 5004
No data returned yet. Trying again id 5004 every 10 seconds
Congrats, your data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\earmmor\AppData\Local\Temp\Rtmp6b251j/test.csv.gz
Rows: 1160 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): start_id, end_id, strategy, json
dbl (3): distance, time_seconds, calories
\
ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
0 routes removed
5004: deleted
Saving tmp/batch_balanced_school_1_with_1160_routes_of_1160_rows.Rds to tmp
Saving tmp/od_school_1_with_1160_of_1160_rows.Rds to tmp
Combining results
Saving outputdata/2023-07-04/routes_max_dist_school_balanced.Rds
Getting the quietest routes for school journeys
2023-07-07 12:42:51 doing batch 1 of 1
Number of rows in batch: 1160
Looking in the file: tmp/batch_quietest_school_1_with_1160_routes_of_1160_rows.Rds
File does not exist
Starting routing
POSTing the request to create and start the job
Job id: 5005
No data returned yet. Trying again id 5005 every 10 seconds
Congrats, your data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\earmmor\AppData\Local\Temp\Rtmp6b251j/test.csv.gz
Rows: 1160 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): start_id, end_id, strategy, json
dbl (3): distance, time_seconds, calories
\
ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
0 routes removed
5005: deleted
Saving tmp/batch_quietest_school_1_with_1160_routes_of_1160_rows.Rds to tmp
Saving tmp/od_school_1_with_1160_of_1160_rows.Rds to tmp
Combining results
Saving outputdata/2023-07-04/routes_max_dist_school_quietest.Rds
Getting the ebike routes for school journeys
2023-07-07 12:44:52 doing batch 1 of 1
Number of rows in batch: 1160
Looking in the file: tmp/batch_ebike_school_1_with_1160_routes_of_1160_rows.Rds
File does not exist
Starting routing
POSTing the request to create and start the job
Job id: 5006
No data returned yet. Trying again id 5006 every 10 seconds
Congrats, your data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\earmmor\AppData\Local\Temp\Rtmp6b251j/test.csv.gz
Rows: 1160 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): start_id, end_id, strategy, json
dbl (3): distance, time_seconds, calories
-
ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
0 routes removed
5006: deleted
Saving tmp/batch_ebike_school_1_with_1160_routes_of_1160_rows.Rds to tmp
Saving tmp/od_school_1_with_1160_of_1160_rows.Rds to tmp
Combining results
Saving outputdata/2023-07-04/routes_max_dist_school_ebike.Rds
• built target r_school [7.986 minutes]
• start target od_data
• built target od_data [1 seconds]
• start target uptake_list_school
Uptake for fastest school routes
✖ error target uptake_list_school
• end pipeline [8.034 minutes]
Warning messages:
1: package 'ggplot2' was built under R version 4.2.3 
2: package 'tibble' was built under R version 4.2.3 
3: package 'dplyr' was built under R version 4.2.3 
4: package 'sf' was built under R version 4.2.3 
Error:
! Error running targets::tar_make()
  Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
  Debugging guide: https://books.ropensci.org/targets/debugging.html
  How to ask for help: https://books.ropensci.org/targets/help.html
  Last error: ℹ In argument: `car_driver_go_dutch = car_driver - (bicycle_increase *
  car_driver/all)`.
ℹ In group 1: `route_id = 21963`.
Caused by error:
! object 'car_driver' not found

```

mem48 commented:
I think this is because I did not include other modes in my data. The HUS data has a different mode split to the commute data and so car driver is not so easily defined.

mem48 commented:
Would be useful to have some documentation on the input variables expected for later steps in the process

mem48 commented:
```
routes = routes %>%
    mutate(
      pcycle_go_dutch = pcycle_go_dutch + (bicycle / all),
      pcycle_go_dutch = case_when(# Prevent the percentage of trips made by bike going above 100%:
        pcycle_go_dutch > 1 ~ 1,
        TRUE ~ pcycle_go_dutch),
      bicycle_go_dutch = pcycle_go_dutch * all,
      bicycle_increase = bicycle_go_dutch - bicycle,
      car_driver_go_dutch = car_driver - (bicycle_increase * car_driver / all),
      car_passenger_go_dutch = car_passenger - (bicycle_increase * car_passenger / all),
      public_transport = train + bus,
      public_transport_go_dutch = public_transport - (bicycle_increase * public_transport / all),
      foot_go_dutch = foot - (bicycle_increase * foot / all),
      other_go_dutch = other - (bicycle_increase * other / all)
    )
```
